### ISSUE TITLE
Bundle archive order

### ISSUE THREAD

**Issue by jemgoss** [getporter/porter#2197]
Due to the unfortunate consequence of `bundle.json` coming after `artifacts` alphabetically, archives always have the `bundle.json` at the end:
```
$ tar tf bun.tgz 
./
./artifacts/
./artifacts/layout/
./artifacts/layout/blobs/
./artifacts/layout/blobs/sha256/
./artifacts/layout/blobs/sha256/032164e3fb4b802e7ad3a2a7aec93babe10572e3c6423ab75f4bf0ade2a74e91
...
./artifacts/layout/index.json
./artifacts/layout/oci-layout
./bundle.json
```
That means you have to trawl through possibly many gigs of archive before you can extract the CNAB bundle manifest in an archive. This affects the performance of `porter publish --archive...` as it first extracts the bundle from the archive, which can take some time.

Also, we're exploring other uses-cases where a client might like to extract some bundle meta-data before uploading the whole archive file and clearly having the `bundle.json` file at the beginning would make life better.

I've explored ways to have `porter archive` place the bundle.json at the beginning by supplying multiple includes in `porter/pkg/porter/archive.go`:
```
	tarOptions := &archive.TarOptions{
		Compression:      archive.Gzip,
		//IncludeFiles:     []string{"."},
		IncludeFiles:     []string{"bundle.json", "artifacts"},
		IncludeSourceDir: true,
	}
	rc, err := archive.TarWithOptions(archiveDir, tarOptions)
```

but this isn't quite what we want, since the paths are all expected to be rooted to "./" and the above yields:
```
bundle.json
artifacts/
artifacts/layout/
...
```

Before going any further, is this a legitimate request? Does it violate the CNAB archive format? I couldn't see any mention of ordering at https://github.com/cnabio/cnab-spec/blob/main/104-bundle-formats.md.

**Comment by carolynvs** [comment]
This is a very reasonable request and doesn't go against the CNAB spec. Porter can fix the order so that the chonky files are at the end.

Did you try using `{"./bundlejson", "./artifacts"}`? I feel like we can probably get that to work by tinkering a bit more.

**Comment by Jemgoss** [comment]
> Did you try using `{"./bundlejson", "./artifacts"}`? I feel like we can probably get that to work by tinkering a bit more.

Yeah, tried that and it doesn't actually make any difference. I'll tinker some more - hoping it doesn't get ugly. If only CNAB had chosen `zartifacts` instead :)
